api4/apitestlib.go

```diff
@@ -149,6 +150,16 @@ func setupTestHelper(dbStore store.Store, searchEngine *searchengine.Broker, ent
 	th.Client = th.CreateClient()
 	th.SystemAdminClient = th.CreateClient()
 
+	// Verify handling of the supported true/false values by randomizing on each run.
+	rand.Seed(time.Now().UTC().UnixNano())
+	trueValues := []string{"1", "t", "T", "TRUE", "true", "True"}
+	falseValues := []string{"0", "f", "F", "FALSE", "false", "False"}
+	trueString := trueValues[rand.Intn(len(trueValues))]
+	falseString := falseValues[rand.Intn(len(falseValues))]
+	mlog.Debug("Configured Client4 bool string values", mlog.String("true", trueString), mlog.String("false", falseString))
+	th.Client.SetBoolString(true, trueString)
+	th.Client.SetBoolString(false, falseString)
+
```

Comment:
Having a randomized test seams like a bit to much "magic" to me. If the bool parsing contains a bug, it's tricky to spot the issue, because a re-test might not fail.

Having just one (table) test for all possible values would be more deterministic. E.g. `getBot` could be used.

Reply:
Agree with both, the risk of having errors that might not reproduce with a rerun, and the solution of clearly pointing at the specific failure covering all of them in one test.

Reply:
The challenge I faced was that I didn't think it was appropriate for any one test to contain the proposed table test -- e.g. testing with `getBot` verifies that /it/ uses `strconv.ParseBool` correctly, but not that anyone else does. 

The intent was to verify that any existing /and future/ API endpoints (that rely on `Client4` to execute the tests) correctly use `strconv.ParseBool`. If I hard-code a particular value (like `0` or `1`), I'm worried a new API endpoint will write the test to match that value (i.e. `newBool == "1"), and then regress on the intent of this functionality.

No disagreement that this is magical, of course. Thoughts on keeping it anyway?

Reply:
I'm fine with merging the PR as it is.

Maybe "real" unit test in `api` will the possible in the future, where we can just test all combinations for all method in little time.

Reply:
I'm fine with merging it as well